### PR TITLE
Enable inlay hints for the supporting servers

### PR DESCRIPTION
**Pull request by VlaDexa**
Since latest neovim nightly supports inlay hints maybe you should look into enabling them by default, like vscode does.

This can be an unwanted change for some people, since the implemented hints are in the middle of the code and I know some people that don't like that.

Maybe that can be left in, but commented out so the users may know that this feature is supported, just not enabled by default.

### CODE REVIEW

**Comment by WindSoilder** [comment]
I think it's good to have a way to toggle inlay hint, because it can be too verbosely

**Comment by feoh** [comment]
Agreed. A toggle or configurable is needed here.

**Comment by VlaDexa** [comment]
Maybe since this affects code stuff and uses LSP it should be <leader>ci?

**Comment by feoh** [comment]
@VlaDexa Thanks for your work thus far. If you're still intrerested in getting this merged, please add a toggle as folks have detailed. I don't honestly care which key bind you use. People can always change it :)

**Comment by WindSoilder** [comment]
Sorry to have more thought here: I don't think `<leader>ci` match the command naming style in kickstart, because `c` doesn't show in help message.

For example:
```lua
map('<leader>ca', vim.lsp.buf.code_action, '[C]ode [A]ction')
map('<leader>rn', vim.lsp.buf.rename, '[R]e[n]ame')
map('<leader>ds', require('telescope.builtin').lsp_document_symbols, '[D]ocument [S]ymbols')
```
Every character has a highlight in help message(which wrapped with `[]`)

**Comment by feoh** [comment]
More thoughts are good so long as the contributor of said thoughts helps find concordance and we can move forward, assuming folks agree that we all want this optional feature.

**Comment by dam9000** [comment]
`<leader>ti` `[T]oggle [I]nlay` still makes the most sense to me :)

**Comment by VlaDexa** [comment]
If we're going by the starting letters of the help message logic, then sure, I agree that `ci` is not conforming.  Then, I think that it being `<leader>th` is easier to remember, since you don't need to know that it toggles something inlay, but enables hints.

**Comment by dam9000** [comment]
`<leader>th` is fine with me :+1:

File: init.lua [hunk 1]
```diff
@@ -286,6 +286,7 @@ require('lazy').setup({
         ['<leader>r'] = { name = '[R]ename', _ = 'which_key_ignore' },
         ['<leader>s'] = { name = '[S]earch', _ = 'which_key_ignore' },
         ['<leader>w'] = { name = '[W]orkspace', _ = 'which_key_ignore' },
+        ['<leader>T'] = { name = '[T]oggle', _ = 'which_key_ignore' },
```
Comment (dam9000) [inline init.lua:289]: `<leader>T` should be lower case here so `<leader>t` to match the `<leader>th` below

Comment (VlaDexa) [reply init.lua]: Thanks!

**Pull request merged**